timers/timers.go

```diff
@@ -0,0 +1,164 @@
+package timers
+
+import (
+	"runtime"
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/uber-go/atomic"
+)
+
+const (
+	cacheline = 64
+
+	// State machine for timers.
+	stateDefault = iota
+	stateExpired
+	stateCanceled
+)
+
+type mutex struct {
+	sync.Mutex
+	// Avoid false sharing:
+	// http://mechanical-sympathy.blogspot.com/2011/07/false-sharing.html
+	pad [cacheline - unsafe.Sizeof(sync.Mutex{})]byte
+}
+
+// Timer FIXME
+type Timer struct {
+	f        func()
+	deadline uint64 // in ticks
+	state    atomic.Int32
+	prev     *Timer
+}
+
+// Stop FIXME
+func (t *Timer) Stop() {
+	t.state.Store(stateCanceled)
+}
+
+func (t *Timer) append(tail *Timer) *Timer {
+	tail.prev = t
+	return tail
+}
+
+// Wheel FIXME
+type Wheel struct {
```

Comment:
My bad on number of buckets, it's definitely a lot more memory to do that. We'll need to use a linked-list like approach, although each list element could store multiple timers.

We need another benchmark that actually involves firing timers, since the optimizations I'm suggesting are optimizing the fired timers path, and the current benchmark only looks at creating+stopping timers.